scripts/core/EXT_Exp_SharedContexts.rst

```diff
@@ -0,0 +1,51 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_experimental_shared_contexts:
+
+=====================================
+ Shared-Contexts Extension
+=====================================
+
+API
+----
+
+* Enumerations
+
+
+    * ${x}_shared_contexts_exp_version_t
+
+
+* Functions
+
+    * ${x}MemGetAllocPropertiesExp
+    * ${x}DriverGetSharedContextExp
+    * ${x}DriverAddContextToSharedExp
+
+Shared Contexts
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Contexts provide a way to isolate resources in an Level Zero application. Specifically,
+USM allocations are created with a context handle and can be used only with
+other objects (like command lists) that have been created with the same context.
+
+To be able to use those resources in applications where USM pointers
+(and not their contexts) are exchanged between multiple libraries, the concept
+of shared contexts interfaces is defined.
+
+A shared context is an Level Zero context handle that encompases a set of contexts
+handles. Any Level Zero object belonging to one of the contexts
+associated with the shared context can be used and interact with an object of the other
+peer contexts. Ownership of L0 objects is not transferred, however.
+
+A given context can only belong to a single shared context, and once created,
+a shared context can be used normally in any API. Objects created using
```

Comment:
I see the cluase that objects created using shared context belong to all context.

What if context is added later to shared context, are those resource allocated prior to this addition are now available in this newly added context?

Reply:
@MichalMrozek . Correct. If user already created a shared-context, it means they have explicitly opted-in for resource sharing, so it is naturally to assume that any resource created with the shared-context can be used with any of the contexts in the set.

Do you think there could be problems with that? if so, then we could just add a prohibition that no resources cannot be created using the shared-context.

Reply:
@MichalMrozek normal contexts would be the typical case, but I dont see why a shared context couldn't be used, but again, if we think there could be problems, we could add the restriction.